syntax = "proto3";

import "header.proto";

enum RobotMode {
    MODE_UNKNOWN = 0;
    MODE_MANUAL = 1; // TODO: Remote Control actions should work only in manual mode
    MODE_GUIDED = 2; // TODO: Goto actions should work only in guided mode
}

enum ArmState {
    ARM_UNKNOWN_STATE = 0;
    ARM_UP = 1;
    ARM_DOWN = 2;
    ARM_STOWED = 3;
}

message Telem {
    Header header = 1;
    float x = 2;
    float y = 3;

    RobotMode mode = 18;

    ArmState leftArmState = 16;
    ArmState rightArmState = 17;
}

message SetModeRequest {
    Header header = 1;
    RobotMode mode = 2;
}

message SetModeResponse {
    Header header = 1;
    bool modeSetSuccess = 2;
}

service TelemService {
    rpc SetMode (SetModeRequest) returns (SetModeResponse) {}
}
